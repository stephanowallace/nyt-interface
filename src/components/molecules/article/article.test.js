import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import Constants from 'Constants';
import Article from '.';

describe('Article', () => {
  const props = {
    abstract: 'abstract test',
    byLine: 'by StÃ©phano',
    loading: false,
    multiSizesImage: [],
    publishDate: moment('2018-12-15T13:22:47-05:00').format('MMM DD, YYYY'),
    title: 'title tst',
    url: 'https://www.thoughtworks.com/',
  };

  it('renders default state', () => {
    const component = shallow(<Article {...props}>test</Article>);
    expect(component).toMatchSnapshot();
    component.instance().renderImage({ mode: Constants.SCREEN_SIZES.MEDIUM });
    expect(component).toMatchSnapshot();
  });

  it('renders expanded content', () => {
    const component = shallow(<Article {...props}>test</Article>);
    component.setState({ isExpanded: true });
    component.instance().renderImage({ mode: Constants.SCREEN_SIZES.SMALL });
    expect(component).toMatchSnapshot();
    component.setState({ isExpanded: false });
    component.instance().renderImage({ mode: Constants.SCREEN_SIZES.EXTRA_SMALL });
    expect(component).toMatchSnapshot();
  });

  it('toggles content', () => {
    const component = shallow(<Article {...props}>test</Article>);
    component.setState({ isExpanded: false });
    component.find('LinkButton').simulate('click');
    expect(component.state().isExpanded).toBeTruthy();
  });

  it('doesnt render image if null', () => {
    const propsWithImage = {
      ...props,
      multiSizesImage: [{
        caption: '',
        copyright: 'Gaia Stella',
        format: 'Standard Thumbnail',
        height: 75,
        subtype: 'photo',
        type: 'image',
        url: 'https://static01.nyt.com/images/2018/12/11/science/11DIET/merlin_147850557_bd693001-7550-4fa6-8219-4ebffcda8e49-thumbStandard.jpg',
        width: 75,
      }],
    };
    const component = shallow(<Article {...propsWithImage}>test</Article>);
    expect(component).toMatchSnapshot();
  });
});
