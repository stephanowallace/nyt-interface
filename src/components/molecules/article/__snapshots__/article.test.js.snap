// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article doesnt render image if null 1`] = `
<div
  className="container"
>
  <ArticlePublishDate
    className="publishDate"
    loading={false}
  >
    Dec 15, 2018
  </ArticlePublishDate>
  <div
    className="content"
  >
    <LinkButton
      className=""
      disabled={false}
      onClick={[Function]}
    >
      <ArticleTitle
        className=""
        loading={false}
      >
        title tst
      </ArticleTitle>
    </LinkButton>
    <ArticleByLine
      className=""
      loading={false}
    >
      by Stéphano
    </ArticleByLine>
  </div>
  <Responsive
    sizes={
      Object {
        "extraSmall": 500,
        "large": 1200,
        "medium": 992,
        "small": 768,
      }
    }
  >
    <Component />
  </Responsive>
</div>
`;

exports[`Article renders default state 1`] = `
<div
  className="container"
>
  <ArticlePublishDate
    className="publishDate"
    loading={false}
  >
    Dec 15, 2018
  </ArticlePublishDate>
  <div
    className="content"
  >
    <LinkButton
      className=""
      disabled={false}
      onClick={[Function]}
    >
      <ArticleTitle
        className=""
        loading={false}
      >
        title tst
      </ArticleTitle>
    </LinkButton>
    <ArticleByLine
      className=""
      loading={false}
    >
      by Stéphano
    </ArticleByLine>
  </div>
  <Responsive
    sizes={
      Object {
        "extraSmall": 500,
        "large": 1200,
        "medium": 992,
        "small": 768,
      }
    }
  >
    <Component />
  </Responsive>
</div>
`;

exports[`Article renders default state 2`] = `
<div
  className="container"
>
  <ArticlePublishDate
    className="publishDate"
    loading={false}
  >
    Dec 15, 2018
  </ArticlePublishDate>
  <div
    className="content"
  >
    <LinkButton
      className=""
      disabled={false}
      onClick={[Function]}
    >
      <ArticleTitle
        className=""
        loading={false}
      >
        title tst
      </ArticleTitle>
    </LinkButton>
    <ArticleByLine
      className=""
      loading={false}
    >
      by Stéphano
    </ArticleByLine>
  </div>
  <Responsive
    sizes={
      Object {
        "extraSmall": 500,
        "large": 1200,
        "medium": 992,
        "small": 768,
      }
    }
  >
    <Component />
  </Responsive>
</div>
`;

exports[`Article renders expanded content 1`] = `
<div
  className="container"
>
  <ArticlePublishDate
    className="publishDate"
    loading={false}
  >
    Dec 15, 2018
  </ArticlePublishDate>
  <div
    className="content"
  >
    <LinkButton
      className=""
      disabled={false}
      onClick={[Function]}
    >
      <ArticleTitle
        className=""
        loading={false}
      >
        title tst
      </ArticleTitle>
    </LinkButton>
    <Paragraph
      className=""
    />
    <Paragraph
      className=""
    >
      abstract test
      <span
        className="externalLink"
      >
        <test-file-stub />
        <Link
          className=""
          href="https://www.thoughtworks.com/"
          target="_blank"
          visited={true}
        >
          See full article
        </Link>
      </span>
    </Paragraph>
    <ArticleByLine
      className=""
      loading={false}
    >
      by Stéphano
    </ArticleByLine>
  </div>
  <Responsive
    sizes={
      Object {
        "extraSmall": 500,
        "large": 1200,
        "medium": 992,
        "small": 768,
      }
    }
  >
    <Component />
  </Responsive>
</div>
`;

exports[`Article renders expanded content 2`] = `
<div
  className="container"
>
  <ArticlePublishDate
    className="publishDate"
    loading={false}
  >
    Dec 15, 2018
  </ArticlePublishDate>
  <div
    className="content"
  >
    <LinkButton
      className=""
      disabled={false}
      onClick={[Function]}
    >
      <ArticleTitle
        className=""
        loading={false}
      >
        title tst
      </ArticleTitle>
    </LinkButton>
    <ArticleByLine
      className=""
      loading={false}
    >
      by Stéphano
    </ArticleByLine>
  </div>
  <Responsive
    sizes={
      Object {
        "extraSmall": 500,
        "large": 1200,
        "medium": 992,
        "small": 768,
      }
    }
  >
    <Component />
  </Responsive>
</div>
`;
